### PR TITLE
Renamed HyperwalletUser to User and interfaces to final classes

### PR DESCRIPTION
**Pull request by azakrevska-epam**
HW-53268 Core HyperwalletUser fix Class

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/model/user/User.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
 /**
  * Represents the User fields.
  */
-public final class HyperwalletUser implements HyperwalletJsonModel, Parcelable {
+public final class User implements HyperwalletJsonModel, Parcelable {
```
Comment (peter-joseph) [inline core/src/main/java/com/hyperwallet/android/model/user/User.java]: I think there will be a task ticket to rename all core to remove `Hyperwallet` prefix, maybe we don't need to do this know, cause we will be forced to fix UI modules in this sprint

File: core/src/main/java/com/hyperwallet/android/model/user/User.java [hunk 2]
```diff
@@ -125,9 +125,9 @@
 
     }
 
-    public interface ProfileTypes {
-        String INDIVIDUAL = "INDIVIDUAL";
-        String BUSINESS = "BUSINESS";
+    public final class ProfileTypes {
```
Comment (peter-joseph) [inline core/src/main/java/com/hyperwallet/android/model/user/User.java]: Please fix all getters that uses this pattern ` return (String) mFields.get(XXX)` for example
```
    @Nullable
    public String getToken() {
        return (String) mFields.get(UserFields.TOKEN);
    }
```

to 
```
    @Nullable
    public String getToken() {
        return getField(UserFields.TOKEN));
    }
```

File: core/src/main/java/com/hyperwallet/android/model/user/User.java (hunk 1 above)
Comment (azakrevska-epam) [reply core/src/main/java/com/hyperwallet/android/model/user/User.java]: @peter-joseph Done

File: core/src/main/java/com/hyperwallet/android/model/user/User.java (hunk 2 above)
Comment (azakrevska-epam) [reply core/src/main/java/com/hyperwallet/android/model/user/User.java]: @peter-joseph Done

**Pull request merged**